### PR TITLE
Allow config generation into empty directory

### PR DESCRIPTION
**Pull request by mniewrzal**
Fixes https://storjlabs.atlassian.net/browse/V3-839

### CODE REVIEW

File: pkg/utils/io.go [hunk 1]
```diff
@@ -48,3 +48,25 @@ func LogClose(fh io.Closer) {
...
+	}
+
+	f, err := os.Open(name)
+	if err != nil {
+		return false, err
+	}
+	defer f.Close()
+
+	_, err = f.Readdir(1)
+	if err == io.EOF {
```
Comment (egonelbre) [inline pkg/utils/io.go]: return err == io.EOF, nil

File: pkg/utils/io.go [hunk 2]
```diff
@@ -48,3 +48,25 @@ func LogClose(fh io.Closer) {
 	}
 	zap.S().Errorf("Failed to close file: %s", err)
 }
+
+// IsValidSetupDirectory checks if directory is valid for setup configuration
+func IsValidSetupDirectory(name string) (bool, error) {
```
Comment (egonelbre) [inline pkg/utils/io.go]: Let's move this into `fpath` and rename it to `fpath.IsEmptyDirectory`.

File: pkg/utils/io.go [hunk 3]
```diff
@@ -48,3 +48,25 @@ func LogClose(fh io.Closer) {
...
+	_, err := os.Stat(name)
+	if err != nil {
```
Comment (egonelbre) [inline pkg/utils/io.go]: This can return an error when you don't have permissions to install to that directory.

File: cmd/satellite/main.go [hunk 4]
```diff
@@ -121,9 +122,9 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
-	_, err = os.Stat(setupCfg.BasePath)
-	if !setupCfg.Overwrite && err == nil {
-		fmt.Println("An satellite configuration already exists. Rerun with --overwrite")
+	valid, err := utils.IsValidSetupDirectory(setupCfg.BasePath)
+	if !setupCfg.Overwrite && !valid {
+		fmt.Printf("An satellite configuration already exists (%v). Rerun with --overwrite\n", setupCfg.BasePath)
```
Comment (egonelbre) [inline cmd/satellite/main.go]: typo "A satellite configuration"

File: cmd/captplanet/setup.go [hunk 5]
```diff
@@ -55,9 +56,9 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
-	_, err = os.Stat(setupCfg.BasePath)
-	if !setupCfg.Overwrite && err == nil {
-		fmt.Println("A captplanet configuration already exists. Rerun with --overwrite")
+	valid, err := utils.IsValidSetupDirectory(setupCfg.BasePath)
+	if !setupCfg.Overwrite && !valid {
+		fmt.Printf("A captplanet configuration already exists (%v). Rerun with --overwrite\n", setupCfg.BasePath)
```
Comment (egonelbre) [inline cmd/captplanet/setup.go]: Remove the prefix "A " here as well to be consistent with other binaries.

File: pkg/utils/io.go (hunk 3 above)
Comment (mniewrzal) [reply pkg/utils/io.go]: yes, that's why I didn't name it `IsEmptyDirectory` from the beginning but maybe the comment was misleading :)

Comment (egonelbre) [reply pkg/utils/io.go]: I mean check the error and don't return true when you don't have permissions (or there is some other error).

**Pull request merged**